Proj isnt compiling in docker container, but does locally

I'm not sure if this is a Proj issue specifically, this might be an `ffi` issue, but I'm hoping you'll have a better hunch than I.

I have a project that compiles locally just fine, and runs fine on an ubuntu based image in CI.
When I try to build a local docker image, I get the following error:
```
#16 236.3 error[E0308]: mismatched types
#16 236.3   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/proj-0.27.0/src/network.rs:53:48
#16 236.3    |
#16 236.3 53 |             let _ = unsafe { CString::from_raw(header.as_ptr() as *mut i8) };
#16 236.3    |                              ----------------- ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*mut u8`, found `*mut i8`
#16 236.3    |                              |
#16 236.3    |                              arguments to this function are incorrect
#16 236.3    |
#16 236.3    = note: expected raw pointer `*mut u8`
#16 236.3               found raw pointer `*mut i8`
#16 236.3 note: associated function defined here
#16 236.3   --> /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/alloc/src/ffi/c_str.rs:397:19
#16 236.3
#16 236.3 For more information about this error, try `rustc --explain E0308`.
#16 236.3 error: could not compile `proj` (lib) due to previous error
```

It seems proj can't compile due to a type mismatch with `CString` in the container.
I'm guessing this an API being called in `proj-sys`? But don't really know where to start debugging this. 😅 

Any advice would be highly appreciated!

Weird, I've tried boiling down by dockerfile to the bare minimum needed for it to reproduce the issue.

Here it is: `Dockerfile`:
```
FROM rust:1.70

WORKDIR /app

# Install the required system dependencies based on what I'm seeing in the georust project: 
# See here: https://github.com/georust/docker-images/blob/main/rust-1.70/libproj-builder.Dockerfile
RUN apt-get update \
    && DEBIAN_FRONTEND="noninteractive" apt-get install -y \
    lld \
    clang \
    libcurl4-gnutls-dev \
    libsqlite3-dev \
    libtiff5-dev \
    cmake \
    pkg-config \
    sqlite3 \
    wget
# Copy all files from our working environment to our Docker image
COPY . .

RUN cargo build --release


ENTRYPOINT ["./target/release/nb-platform"]
```

And this gives me the above error:
```
#9 224.7 error[E0308]: mismatched types
#9 224.7   --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/proj-0.27.0/src/network.rs:53:48
#9 224.7    |
#9 224.7 53 |             let _ = unsafe { CString::from_raw(header.as_ptr() as *mut i8) };
#9 224.7    |                              ----------------- ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*mut u8`, found `*mut i8`
#9 224.7    |                              |
#9 224.7    |                              arguments to this function are incorrect
#9 224.7    |
#9 224.7    = note: expected raw pointer `*mut u8`
#9 224.7               found raw pointer `*mut i8`
#9 224.7 note: associated function defined here
#9 224.7   --> /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/alloc/src/ffi/c_str.rs:397:19
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+- Bump geo-types
+- Allow typeck to infer pointer types
 - Bump `approx` dev dependency to match `geo-types`. This doesn't affect
   downstream users, only those building the proj crate for development.
   -  <https://github.com/georust/proj/pull/138>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ rust-version = "1.63"
 
 [dependencies]
 proj-sys = { version = "0.23.1", path = "proj-sys" }
-geo-types = { version = "0.7.8", optional = true }
+geo-types = { version = "0.7.10", optional = true }
 libc = "0.2.119"
 num-traits = "0.2.14"
 thiserror = "1.0.30"
@@ -33,7 +33,7 @@ network = ["reqwest", "proj-sys/network"]
 [dev-dependencies]
 # approx version must match the one used in geo-types
 approx = ">= 0.4.0, < 0.6.0"
-geo-types = { version = "0.7.3", features = ["approx"] }
+geo-types = { version = "0.7.10", features = ["approx"] }
 
 [package.metadata.docs.rs]
 features = [ "proj-sys/nobuild", "network", "geo-types" ]

--- a/src/network.rs
+++ b/src/network.rs
@@ -57,7 +57,7 @@ impl Drop for HandleData {
     // dereferencing it if need be so the resource is freed
     fn drop(&mut self) {
         if let Some(header) = self.hptr {
-            let _ = unsafe { CString::from_raw(header.as_ptr().cast::<i8>()) };
+            let _ = unsafe { CString::from_raw(header.as_ptr().cast()) };
         }
     }
 }
@@ -192,7 +192,7 @@ unsafe fn _network_open(
     // Copy the downloaded bytes into the buffer so it can be passed around
     res.bytes()?
         .as_ptr()
-        .copy_to_nonoverlapping(buffer.cast::<u8>(), contentlength.min(size_to_read));
+        .copy_to_nonoverlapping(buffer.cast(), contentlength.min(size_to_read));
     let hd = HandleData::new(url, headers, None);
     // heap-allocate the struct and cast it to a void pointer so it can be passed around to PROJ
     let hd_boxed = Box::new(hd);

